package.json

```
@@ -38,7 +40,7 @@
     "changelog:from-beginning": "conventional-changelog -p @lmc-eu/conventional-changelog-lmc-bitbucket -i CHANGELOG.md -s -r 0"
   },
   "dependencies": {
-    "@lmc-eu/spirit-design-tokens": "^0.2.0",
+    "@lmc-eu/spirit-design-tokens": "~0.4.0",
```

```suggestion
    "@lmc-eu/spirit-design-tokens": "^0.4.0",
```

https://developer.happyr.com/always-use-caret-instead-of-tilde

---

I disagree. `0.x` versions mark an [early stage of development](https://semver.org/#spec-item-4) with breaking changes happening even during minor version bumps. Allowing upgrade to a higher minor version of `0.x` thus may break the app.

Once the referred package version is `>=1`, then it's time to change `~` to `^`.

---

> Allowing upgrade to a higher minor version of 0.x thus may break the app.

But `^0.4.0` does not allow this. It means `>=0.4.0  < 0.5.0`, so "do safe upgrades without BC". So in the pre-1.x version, the behavior is effectively the same as using tilde. However after 1.x, the caret `^` follows safe semver upgrades (unlike tilde, which behaves differently and `~1.2.3` won't upgrade to `1.3.0`) .

So exactly what you wrote is the reason why use caret, for both 0.x and 1.x+ versions, without a need to remember you should change it after 1.x.

https://github.com/npm/node-semver#caret-ranges-123-025-004
https://semver.npmjs.com/

---

@OndraM @adamkudrna And what about to pin the version to exact match, eg. `"@lmc-eu/spirit-design-tokens": "0.4.0",` and let dependabot to take care of updates?

Everybody got the same version and breaking and other changes would be under control in separate PRs. And also there will be no arguing about tilda or caret ;-)

---

> So in the pre-1.x version, the behavior is effectively the same as using tilde.

Nice one, didn't know that!